### PR TITLE
added metadata overwrite ability

### PR DESCRIPTION
**Pull request by rubahatch**
added extraction of Content-Type and Content-Disposition from metadata and  added them to the upload input for s3. 
I kept everything else as is- also those keys were not removed from metadata structure so on upload they will have "x-amz-meta" prefix but that won't affect anything.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,13 @@
 module github.com/graymeta/stow
 
+go 1.14
+
 require (
 	github.com/Azure/azure-sdk-for-go v32.5.0+incompatible
-	github.com/Azure/go-autorest/autorest v0.9.0 // indirect
+	github.com/Azure/go-autorest/autorest/to v0.3.0 // indirect
```
Comment (8tomat8) [inline go.mod:7]: I think there was a reason to use the newer version. Not like we need it anyway, but if we do... I do not want to debug this :rofl:

File: s3/container.go [hunk 2]
```diff
@@ -266,3 +268,24 @@ func parseMetadata(md map[string]*string) (map[string]interface{}, error) {
...
+type toOverwrite struct {
+	contentType *string
+	contentDisposition *string
+}
+
+//todo: clean metadata from overwrites
+func metadataOverwrite(md map[string]*string) toOverwrite {
+	var to toOverwrite
+	for key, value := range md {
+		switch key {
```
Comment (8tomat8) [inline s3/container.go:281]: `ToLower`

File: s3/container.go [hunk 3]
```diff
@@ -266,3 +268,24 @@ func parseMetadata(md map[string]*string) (map[string]interface{}, error) {
 	}
 	return m, nil
 }
+
+type toOverwrite struct {
+	contentType *string
+	contentDisposition *string
+}
+
+//todo: clean metadata from overwrites
```
Comment (8tomat8) [inline s3/container.go:277]: Add a comment on why we need to do it. And why it is `Overwrite`.

File: s3/container.go (hunk 2 above)
Comment (rubahatch) [reply s3/container.go:281]: for flexibility?

Comment (8tomat8) [reply s3/container.go:281]: Yep. So it just works and we do not need to document it anywhere.

**Pull request merged**